### PR TITLE
Added Factorial in SCALA

### LINKED ISSUE #2098: Add Factorial in Scala
This Issue requires Factorial Program to Be Added To The Scala Repository

### PR DESCRIPTION
**Pull request by uditpatel2526**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! 
For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [x] I fixed #2098 

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [x] I added fun facts (i.e. debut developer, typing, etc.)
  - [x] I added reference link(s) to the README
  - [x] I added solution citations when necessary (see [plagiarism][2])

### Testing

- [x] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [x] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism

### CODE REVIEW

**Review by jrg94 — CHANGES_REQUESTED** [review]
Looks good! Just make sure the program pulls from the command line.

**Comment by uditpatel2526** [comment]
Hey @jrg94, thanks for suggesting changes. I am new to scala. If you can just tell me what do  I do to make sure the program gets merged, it'll be really great.

File: archive/c/c-plus-plus/linear-search.CPP [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// C++ code to linearly search x in arr[]. If x
```
Comment (jrg94) [inline archive/c/c-plus-plus/linear-search.CPP:1]: Can we remove this file for this pull request?

File: archive/s/scala/factorial.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+		if (n == 0) 
+			return 1
+		else
+			return n * factorial(n-1) 
+	} 
+
+	// Driver Code 
+	def main(args: Array[String]) 
+	{ 
+		println(factorial(5))
```
Comment (jrg94) [inline archive/s/scala/factorial.scala:21]: Can this pull from the command line (`args`) instead of hardcoding 5?

**Review by jrg94 — CHANGES_REQUESTED** [review]
Hope this helps!

File: archive/c/c-plus-plus/linear-search.CPP (hunk 1 above)
Comment (uditpatel2526) [reply archive/c/c-plus-plus/linear-search.CPP:1]: Yes I will remove that.

File: archive/s/scala/factorial.scala (hunk 2 above)
Comment (uditpatel2526) [reply archive/s/scala/factorial.scala:21]: okay I'll get back to in some time with the updated code.

**Comment by uditpatel2526** [comment]
Hey @jrg94, I have updated the code. Please go through it and let me know if you want any changes.

**Review by jrg94 — APPROVED** [review]
Looks better! We may revisit this when we do some testing. There are deprecation warnings, and the formatting is a bit messy.

**Comment by uditpatel2526** [comment]
Hey @jrg94, If you can add the hacktoberfest-accepted label to this pull request, it'll be really great.

**Comment by jrg94** [comment]
Do you have any documentation on that label? I haven't seen anything about it on the official site.

**Comment by jrg94** [comment]
Nevermind, I saw this: https://hacktoberfest.digitalocean.com/hacktoberfest-update

**Comment by jrg94** [comment]
It's sort of a pain to label all accepted pull requests. Based on the article above, merging and accepting should be enough.

**Comment by jrg94** [comment]
Hey @uditpatel2526! If you enjoyed contributing to this project and would like to make sure that it continues to be maintained, we can always use your support at The Renegade Coder. Here are some ways you can help out:

- Subscribe to the [monthly newsletter](https://newsletter.therenegadecoder.com/)
- Become a [patron](https://www.patreon.com/TheRenegadeCoder)
- Join us on [Discord](https://discord.gg/Jhmtj7Z)
- And, check out the rest of the [list](https://therenegadecoder.com/blog/ways-you-can-help-grow-the-renegade-coder/) which includes links to The Renegade Coder [YouTube channel](https://www.youtube.com/channel/UCpyoVwOqYRlSAEUPEn7P9hw) and [Twitter account](https://twitter.com/RenegadeCoder94)

Thanks for your help!

**Pull request merged**